### PR TITLE
docs: update form example security

### PR DESCRIPTION
**Pull request by WestonThayer**
Context in #2860.

Questions for reviewers:

- Does adding `method="post"` require explanation?
- Should the comment cite the GitHub issue?

If you agree with this approach, I'll look through the rest of the docs and also update the `Form` component to set `method="post"` by default.

### CODE REVIEW

**Comment by johnrom** [comment]
I don't think we need to document specific security concerns or cite github issues in comments. If we need to describe in detail how to avoid security issues with Forms (that's not necessarily our place, as that info will not be nearly complete), we can create an FAQ area related to form security.

Instead, I think we might want to avoid using a username / password form as an example -- it sets a bad precedent to begin with. Authentication-related forms require many well-thought-out layers, and probably shouldn't be oversimplified like this.

**Comment by WestonThayer** [comment]
I agree with that approach. Do you want me to update this PR to swap username/password for something benign like favorite food/favorite dessert?

A FAQ on form security would be great. Do you want to track in a GitHub issue?

**Comment by johnrom** [comment]
I have no problem keeping some authoritative info related to form security in the docs somewhere. But we should link to something like OWASP and encourage users to read up fully on security because when things aren't as benign as favorite ice cream, form stuff is very serious. Even things that seem benign can be very serious, maybe your favorite ice cream is a security question on your credit card password reset form.

https://owasp.org/www-project-top-ten/

**Comment by johnrom** [comment]
I like the idea of switching the form to a "benign" form, btw, anything is better than user / pass.

**Comment by WestonThayer** [comment]
I think both some authoritative resources and formik-specific tips are good (unless there's an authoritative resource out that that talks about the dangers of writing a JS-only form + `<form>`'s default behavior, I couldn't find one).

File: docs/src/pages/docs/overview.md [hunk 1]
```diff
@@ -145,7 +145,10 @@ const Basic = () => (
         isSubmitting,
         /* and other goodies */
       }) => (
-        <form onSubmit={handleSubmit}>
+        <form
+          onSubmit={handleSubmit}
+          method="post" /* Don't leak password if JS disabled (#2860) */
```
Comment (maddhruv) [inline docs/src/pages/docs/overview.md:150]: Can we remove this comment and reference to the issue

**Review by maddhruv — CHANGES_REQUESTED** [review]
Please remove the comment

**Comment by jaredpalmer** [comment]
Agree. Let’s remove the username password example

**Pull request closed without merging**